main/src/main/scala/sbt/nio/Settings.scala

```diff
@@ -93,15 +94,15 @@ private[sbt] object Settings {
           addTaskDefinition(Def.setting[Task[Seq[Path]]](key, init, setting.pos)) ::
             outputsAndStamps(taskKey)
         }
-        ak.manifest.typeArguments match
-          case (t: Manifest[_]) :: Nil if seqClass.isAssignableFrom(t.runtimeClass) =>
-            t.typeArguments match {
-              case p :: Nil if pathClass.isAssignableFrom(p.runtimeClass) => mkSetting[Seq[Path]]
-              case _                                                      => default
-            }
-          case (t: Manifest[_]) :: Nil if pathClass.isAssignableFrom(t.runtimeClass) =>
-            mkSetting[Path]
-          case _ => default
+        if seqClass.isAssignableFrom(tag.typeArg) then
+          // TODO fix this: maybe using the taskKey macro to convey the information
+          // t.typeArguments match {
+          //   case p :: Nil if pathClass.isAssignableFrom(p.runtimeClass) => mkSetting[Seq[Path]]
+          //   case _                                                      => default
+          // }
```

Comment:
This is probably broken since https://github.com/sbt/sbt/pull/6746 and the introduction of `coerceTag`.